frontend/packages/console-shared/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -81,6 +82,7 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
   constructor(props) {
     super(props);
     this.state = {
+      resources: {},
```

Comment:
If the item is already loaded coming into the component, you have a chance of not setting it (ever or at least not immediately). I think you need to bring your logic from `UNSAFE_componentWillReceiveProps` here too (see the `items` line below this one).